### ISSUE TITLE
[BERT/PyTorch] gradient accumulation scaling

### ISSUE THREAD

**Issue by Hannibal046** [NVIDIA/DeepLearningExamples#1112]
Hi, as suggested here https://pytorch.org/docs/stable/notes/amp_examples.html should the loss be divided	by `args.gradient_accumulation_steps`?
https://github.com/NVIDIA/DeepLearningExamples/blob/717c78e9c150742147153824c10ab6e37997552c/PyTorch/LanguageModeling/BERT/run_pretraining.py#L513

```
loss = loss / args.gradient_accumulation_steps
grad_scaler.scale(loss).backward() 
```

**Comment by sharathts** [comment]
https://github.com/NVIDIA/DeepLearningExamples/blob/717c78e9c150742147153824c10ab6e37997552c/PyTorch/LanguageModeling/BERT/run_pretraining.py#L450-L464

Instead of dividing loss by `args.gradient_accumulation_steps` we divide summed gradient by `args.gradient_accumulation_steps` which is the same in theory. The former provides more numerical stability.
